### PR TITLE
Multiple user problem in async await 

### PR DESCRIPTION
**Pull request by Dhroov7**

### CODE REVIEW

File: src/passport/strategies/user/strategy-facebook.js [hunk 1]
```diff
@@ -24,83 +24,90 @@ module.exports = new FacebookStrategy({
...
-                if (fbaccount) {
-                    throw new Error('Your Facebook account is already linked with codingblocks account Id: ' + fbaccount.dataValues.userId)
+        if(userFacebook){
+            if(req.isAuthenticated()){
+                let Ids = ''
+                for(let i=0;i<userFacebook.length;i++){
+                    Ids += userFacebook[i].dataValues.id
+                    if(i<userFacebook.length-1){
+                        Ids += ','
+                    }
```
Comment (championswimmer) [inline src/passport/strategies/user/strategy-facebook.js]:
```js
let oldIds = userFacebook.map((uf) => uf.id).join(',')
```

File: src/passport/strategies/user/strategy-facebook.js [hunk 2]
```diff
@@ -24,83 +24,90 @@ module.exports = new FacebookStrategy({
...
-                                    facebookId: profileJson.id
-                                })
-                                span.finish()
-                            })
-                            return cb(null, user.get())
-                        })
-                        .catch((err) => {
-                            Raven.captureException(err)
-                            return cb(err, null)
+                return cb(null,false,{message:'Sorry,this facebook account is connected with another coding blocks account: ' + Ids})
```
Comment (championswimmer) [inline src/passport/strategies/user/strategy-facebook.js]: `oldIds`

File: src/passport/strategies/user/strategy-facebook.js (hunk 1 above)
Comment (championswimmer) [reply src/passport/strategies/user/strategy-facebook.js]: Variable names should always start with small letter

File: src/passport/strategies/user/strategy-github.js [hunk 3]
```diff
@@ -20,69 +20,83 @@ module.exports = new GithubStrategy({
...
+        })
+
+        if(userGithub){
+            if(req.isAuthenticated()){
+                let Ids = ''
+                for(let i=0;i<userGithub.length;i++){
+                    Ids += userGithub[i].dataValues.id
+                    if(i<userGithub.length-1){
+                        Ids += ','
+                    }
```
Comment (championswimmer) [inline src/passport/strategies/user/strategy-github.js]:
```js
let oldIds = userGithub.map((uf) => uf.id).join(',')
```

File: src/passport/strategies/user/strategy-google.js [hunk 4]
```diff
@@ -14,67 +14,83 @@ module.exports = new GoogleStrategy({
...
+                }
+            })
+
+            if(userGoogle){
+                if(req.isAuthenticated()){
+                    let Ids = ''
+                    for(let i=0;i<userGoogle.length;i++){
+                        Ids += userGoogle[i].dataValues.id
+                        if(i<userGoogle.length-1){
+                            Ids += ','
```
Comment (championswimmer) [inline src/passport/strategies/user/strategy-google.js]: map + join here too

**Comment by Dhroov7** [comment]
@championswimmer Done with all the changes.

**Comment by championswimmer** [comment]
well this did not work **at all** pretty flawed

**Comment by championswimmer** [comment]
we need to find those users 
- with same email
- **but without facebook already connected**

otherwise this simply blocks all facebook logins, for all possible cases (even for legit people).

**Pull request merged**